tests/suites/test_suite_psa_crypto_se_driver_hal.data

```diff
@@ -26,3 +26,72 @@ register_twice:3
 
 Register SE driver: maximum number of drivers
 register_max:
+
+SE key import-export (p_allocate allows all slots)
+key_creation_import_export:0:0
+
+SE key import-export (p_allocate allows 1 slot)
+key_creation_import_export:ARRAY_LENGTH( ram_slots ) - 1:0
+
+SE key import-export, check after restart (slot 0)
+key_creation_import_export:0:1
+
+SE key import-export, check after restart (slot 3)
+key_creation_import_export:3:1
+
+Key creation smoke test: AES-CTR
```

Comment:
Please avoid smoking and say what the test does. The indirection through some new "smoke" concept doesn't help clarity, although it is cute.

If the "smoke" concept includes many other tests, please split the test up into smaller ones that can have clear names. One big long name saying all the tests that "smoke" does wouldn't be helpful.

Alternatively, @AndrzejKurek and @dgreen-arm may replace all this smoke testing as temporary and replace it with mocks that check for expected status codes instead of just "valid" status codes. In which case, do nothing!

Reply:
If you have a better name, please suggest. The goal of this test is to call lots of functions on the key and check that the system doesn't do anything crazy. I think this test will still have value even when there are tests for the actual intended behavior: it's easy to check that it's exhaustive (in the sense of calling all API functions that take a key as input).

Reply:
Let's go with the alternative and use full-coverage mock-based with better names in a follow up PR. Smoking allowed in this PR.